nilearn/datasets.py

```diff
@@ -26,6 +25,17 @@
 
 import nibabel
 
+# Define _get_time, which is used module-wide to get the current time.
+try:
+    # Use uptime (when available) to avoid reporting time while computer slept.
+    import uptime
+    if uptime.uptime() is None:
+        raise Exception("uptime is not supported by this OS.")
+    _get_time = uptime.uptime
```

Comment:
> Do we have a solution that can be coded fast and solve this problem ?

Implement a residual time based on a local speed of download. Or simply
move on.